python/test/unit/runtime/test_jit.py

```diff
@@ -40,3 +40,67 @@ def my_hook(*args, **kwargs):
     out = torch.zeros_like(x)
     with pytest.raises(Exception):
         add_kernel[(4, )](x, y, out, 4, 4)
+
+
+def test_pre_call_hooks_can_stop_execution(device):
+    @triton.jit
+    def add_kernel(
+        in_ptr0,
+        in_ptr1,
+        out_ptr,
+        n_elements,
+        BLOCK_SIZE: "tl.constexpr",
+    ):
+        pid = tl.program_id(axis=0)
+        block_start = pid * BLOCK_SIZE
+        offsets = block_start + tl.arange(0, BLOCK_SIZE)
+        mask = offsets < n_elements
+        x = tl.load(in_ptr0 + offsets, mask=mask)
+        y = tl.load(in_ptr1 + offsets, mask=mask)
+        output = x + y
+        tl.store(out_ptr + offsets, output, mask=mask)
+
+    called = 0
+
+    def my_hook(kernel, *args, **kwargs):
+        nonlocal called
+        called += 1
+        return True
+
+    add_kernel.add_pre_run_hook(my_hook)
```

Comment:
Hmm after some more testing: these hooks aren't sufficient for our use case. We would want something more like a "global hook" that gets attached on all triton kernels, and we'd want to activate this when we're doing tracing in PyTorch. I'll modify this PR in a couple of days (unless you think this is unreasonable, @Jokeren)

Reply:
I'd like to see a brief doc if any.

Hooks are generally fine to me, but we may not want too many

Reply:
Sounds good, I'll write up a quick doc

Reply:
Here you go: https://docs.google.com/document/d/1lMoUdxiPcpiMLN9Aalg_u0nrfZxFGG_F3dsyJt3fc1k/edit

Reply:
Thanks for the feedback on the document. As a status update, I'm planning to test a version of this with PyTorch first and seeing how well it works and finally coming back to update this PR once everything goes well. This will probably take a couple of weeks, so I'll be back then.